components/forms/createMemberForm.tsx

```diff
@@ -0,0 +1,318 @@
+        dob: "06/15/2005",
+        dod: "",
+        spouse: {
+          firstName: "Camila",
+          lastName: "Gomez",
+          relation: "Spouse",
+          dob: "07/20/2006",
+          dod: "",
+        },
+        children: [
+          {
+            id: "9",
+            firstName: "Sofia",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "01/10/2020",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "10",
+            firstName: "Mateo",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "02/20/2020",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "11",
+            firstName: "Valentina",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "03/30/2020",
+            dod: "",
+            children: [],
+          },
+        ],
+      },
+      {
+        id: "2",
+        firstName: "Alejandro",
+        lastName: "Santos",
+        relation: "Child",
+        dob: "08/05/2007",
+        dod: "",
+        spouse: {
+          firstName: "Lucia",
+          lastName: "Lopez",
+          relation: "Spouse",
+          dob: "09/10/2008",
+          dod: "",
+        },
+        children: [],
+      },
+      {
+        id: "3",
+        firstName: "Carmen",
+        lastName: "Santos",
+        relation: "Child",
+        dob: "09/12/2010",
+        dod: "",
+        children: [
+          {
+            id: "4",
+            firstName: "Nicolas",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "04/15/2015",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "5",
+            firstName: "Renata",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "05/25/2016",
+            dod: "",
+            children: [],
+          },
+          {
+            id: "6",
+            firstName: "Alejandro",
+            lastName: "Santos",
+            relation: "Grandchild",
+            dob: "06/30/2017",
+            dod: "",
+            children: [],
+          },
+        ],
+      },
+    ],
+  });
+
+  const [formData, setFormData] = useState({
+    parentId: "",
+    firstName: "",
+    lastName: "",
+    dob: "",
+    dod: "",
+    hasSpouse: false,
+    spouseName: "",
+    spouseDob: "",
+    spouseDod: "",
+  });
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    const { name, value, type } = e.target;
+
+    if (type === "checkbox" || type === "radio") {
+      setFormData({
+        ...formData,
+        [name]: (e.target as HTMLInputElement).checked,
+      });
+    } else {
+      setFormData({
+        ...formData,
+        [name]: value,
+      });
+    }
+  };
+
+  const getAllFamilyMembers = (members: FamilyMember): FamilyMember[] => {
```

Comment:
We need an edge case here. Imagine that a user is creating relationships by dragging and dropping. If they create a relationship like this: Parent1 > Child1> Grandchild1 > Parent1 (again) this would basically create an infinite loop and give us a `runtime error`. 

![image](https://github.com/user-attachments/assets/e7471fc7-9b8f-4d69-9ff1-fc19dd19d786)

I am not really sure how we can prevent it 😅 but we need to have that edge case somewhere

Reply:
Also, I swear I selected different lines lol